ChronoVisor/src/ChronicleMetaDirectory.cpp

```diff
@@ -175,20 +376,40 @@ uint64_t ChronicleMetaDirectory::playback_event(uint64_t sid) {
     return 0;
 }
 
-std::string ChronicleMetaDirectory::get_chronicle_attr(std::string &name,
-                                                       const std::string& key) {
+int ChronicleMetaDirectory::get_chronicle_attr(std::string& name, const std::string& key, std::string& value) {
     LOGD("getting attributes key=%s from Chronicle name=%s", key.c_str(), name.c_str());
     uint64_t cid = CityHash64(name.c_str(), name.size());
-    Chronicle *pChronicle = chronicleMap_->at(cid);
-    return pChronicle->getPropertyList().at(key);
+    Chronicle *pChronicle;
+    std::lock_guard<std::mutex> lock(g_chronicleMetaDirectoryMutex_);
+    try {
+        pChronicle = chronicleMap_->at(cid);
```

Comment:
Remove this. We can add exceptions after a discussion about error handling later.